File: src/main/java/com/ai2dev/sb_lambda_testng_2/dto/CardholderRequest.java
package com.ai2dev.sb_lambda_testng_2.dto;

import com.ai2dev.sb_lambda_testng_2.model.CardholderType;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public record CardholderRequest(
        @NotNull
        @Size(min = 1, max = 15)
        String cardholderTitle,

        @NotNull
        @Size(min = 1, max = 100)
        String cardholderForename,

        @NotNull
        @Size(min = 1, max = 100)
        String cardholderSurname,

        @NotNull
        CardholderType cardholderType,

        @NotNull
        @Size(min = 1, max = 100)
        String cardholderPhotoName,

        @Size(min = 39, max = 42)
        String cardholderPhotoRRSKey,

        @Size(min = 20, max = 2048)
        @Pattern(regexp = "^(https?://).*")
        String cardholderPhotoURL,

        @Size(min = 39, max = 42)
        String cardholderPhotoKey
) {}

File: Models/Requests/CardholderRequest.cs
namespace azuresharpapi153.Models.Requests;

public sealed class CardholderRequest
{
    public string CardholderTitle { get; set; } = string.Empty;
    public string CardholderForename { get; set; } = string.Empty;
    public string CardholderSurname { get; set; } = string.Empty;
    public string CardholderType { get; set; } = string.Empty;
    public string CardholderPhotoName { get; set; } = string.Empty;
    public string? CardholderPhotoRRSKey { get; set; }
    public string? CardholderPhotoURL { get; set; }
    public string? CardholderPhotoKey { get; set; }
}